LoopUp.Siesta/Patch/JsonPatchDocumentHelpers.cs

```diff
@@ -0,0 +1,164 @@
+namespace LoopUp.Siesta.Patch
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using LoopUp.Siesta.Extensions;
+    using Microsoft.AspNetCore.JsonPatch;
+    using Newtonsoft.Json;
+
+    /// <summary>
+    /// Helper class for making and working with JsonPatchDocuments.
+    /// </summary>
+    public static class JsonPatchDocumentHelpers
+    {
+        /// <summary>
+        /// Generate a JsonPatchDocument through the comparison of two versions of the same class.
+        /// </summary>
+        /// <param name="original">Original version of the class.</param>
+        /// <param name="modified">New version of the class.</param>
+        /// <typeparam name="T">The type of the class.</typeparam>
+        /// <returns>A JsonPatchDocument for the two objects.</returns>
+        public static JsonPatchDocument GeneratePatchDocument<T>(T original, T modified)
+            where T : class
+        {
+            var patchDocument = new JsonPatchDocument();
+
+            PopulatePatchForObject(original, modified, patchDocument, string.Empty);
+
+            return patchDocument;
+        }
+
+        private static void PopulatePatchForObject<T>(T? originalObject, T? modifiedObject, JsonPatchDocument patchDocument, string path)
```

Comment:
I think this looks pretty good but it was a bit of a struggle going through it and trying to figure out what was going on - I think some detailed documentation is in order through these methods.

Reply:
Yes that is very fair haha